optimum/gptq/utils.py

```diff
@@ -0,0 +1,94 @@
+#  Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from logging import getLogger
+from typing import Optional, Union
+
+import torch
+from torch import nn
+from transformers.pytorch_utils import Conv1D
+
+from .constants import BLOCK_PATTERNS, SEQLEN_KEYS_TRANFORMERS
+
+
+logger = getLogger(__name__)
+
+
+def get_module_by_name_prefix(model: nn.Module, module_name: str):
+    for name, module in model.named_modules():
+        if name.startswith(module_name):
+            return module
+
+
+def get_layers(module: nn.Module, layers=[Conv1D, nn.Conv2d, nn.Linear], prefix: Optional[str] = None, name: str = ""):
```

Comment:
Can you add a docstring?

IMO vs recursive function it is also more readable to iterate over `name_modules()` and use recursive getattr if needed: https://github.com/huggingface/optimum/blob/e72901cc3937d6bb647e8a63648368da9b560759/optimum/utils/modeling_utils.py#L18

Reply:
Replaced and added the docstring